### PR TITLE
ouster_sdk: add a new package

### PR DESCRIPTION
**Pull request by valgur**
Adds a new recipe for the Ouster SDK: https://github.com/ouster-lidar/ouster_example

The project already contains a conanfile.py and this recipe is largely based on it. The only major difference is that I opted not to package and auto-load the https://github.com/ouster-lidar/ouster_example/blob/master/cmake/OusterSDKConfig.cmake.in file and the generated `OusterSDKTargets.cmake` file. These both are superseded by Conan 2 generators and cpp_info components.

### CODE REVIEW

**Comment by valgur** [comment]
@uilianries `ERROR: Missing prebuilt package for 'libcurl/8.2.1'`

**Comment by valgur** [comment]
@uilianries `ERROR: Missing prebuilt package for 'libtins/4.5'`

**Comment by uilianries** [comment]
I just started an internal build to generate libtins.

**Review by RubenRBS — APPROVED** [review]

File: recipes/ouster_sdk/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import get, copy, rmdir, rm, save, replace_in_file, export_conandata_patches, apply_conandata_patches
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.60.0 <2.0 || >=2.0.6"
+
+class PackageConan(ConanFile):
+    name = "ouster_sdk"
+    description = "Ouster SDK - tools for working with Ouster Lidars"
+    license = "BSD 3-Clause"
```
Comment (jwillikers) [inline recipes/ouster_sdk/all/conanfile.py:16]:
```suggestion
    license = "BSD-3-Clause"
```

**Review by jwillikers — APPROVED** [review]

File: recipes/ouster_sdk/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+        if self.options.build_osf:
+            # Used in fb_generated/*.h
+            self.requires("flatbuffers/24.3.7", transitive_headers=True)
+            self.requires("libpng/[>=1.6 <2]")
+            self.requires("zlib/[>=1.2.11 <2]", transitive_libs=True)
+
+        if self.options.build_viz:
+            self.requires("glad/0.1.36")
+            self.requires("CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS/3.4")
```
Comment (ericLemanissier) [inline recipes/ouster_sdk/all/conanfile.py:100]: @valgur what's that?

Comment (valgur) [reply recipes/ouster_sdk/all/conanfile.py:100]: Ooof. :grin: Thanks! Opened a PR: #23335.

**Pull request merged**